tel2puml/sequence_otel_v2.py

```diff
@@ -85,4 +85,81 @@ def sequence_events_by_async_event_types(
         else:
             non_async_groups.append([event])
     groups = list(async_groups.values()) + non_async_groups
-    return order_groups_by_start_timestamp(groups)
+    return groups
+
+
+def sequence_otel_event_ancestors(
+    event: OTelEvent,
+    event_id_to_event_map: dict[str, OTelEvent],
+    previous_event_ids: list[str] | None = None,
+    async_flag: bool = False,
+    event_to_async_group_map: dict[str, dict[str, str]] | None = None,
+) -> dict[str, list[str]]:
+    """Sequence OTel event ancestors using async information, if available.
+
+    :param event: An OTelEvent.
+    :type event: :class:`OTelEvent`
+    :param event_id_to_event_map: A dictionary mapping event IDs to OTelEvents.
+    :type event_id_to_event_map: `dict`[`str`, :class:`OTelEvent`]
+    :param previous_event_ids: A list of previous event IDs.
+    :type previous_event_ids: `list`[`str`] | `None`
+    :param async_flag: A flag indicating whether to use async information.
```

Comment:
should be flag indicating whether to sequence groups with synchronous or asynchronous logic